KlockanAPI/src/KlockanAPI.Application/CrossCutting/FoundException.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace KlockanAPI.Application.CrossCutting;
+
+public class FoundException : ArgumentException
+{
+    public FoundException(string? message) : base(message) { }
+
+    public static void ThrowIfNotNull<T>(T? param, string? message = null)
```

Comment:
Because this exception is used for example in case you want to delete a course but it is being used by a classroom. Then in the service when looking for the classrooms that are associated to the course, the exception is thrown in case it exists.